h5py/_hl/dataset.py

```diff
@@ -497,6 +577,14 @@ def __iter__(self):
             yield self[i]
 
     @with_phil
+    def iter_chunks(self, sel=None):
+        """ Return chunk iterator.  TypeError if the dataset is not chunked.
+            If sel is provided, only chunks that intersect the selected region
+            will bee reeturned, otherwise all chunks in the dataset will be
```

Comment:
A few things I think the docstring should make clearer:

- What does it yield (tuples of slices)
- `sel` should be a slice or a tuple of slices, such as you can get using `numpy.s_`.
- If you pass a selection, you only get back the intersecting parts of chunks, not the whole of any chunk that intersects the selection.
- This only works for chunked datasets (for now, anyway)

Reply:
updated.